### PR TITLE
Add travis testing support for python 3.8

### LINKED ISSUE #471: Python 3.8 testing in Tox and Travis
Like the title says.

### PR DESCRIPTION
**Pull request by neozenith**
Working through adding python3.8 testing support and fixing any broken tests.

Closes #471 

Could use advice on tidying up the test matrix in the travis.yml file

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,6 +34,27 @@ matrix:
     - env: TOX_SUFFIX="aiohttp"
       python: 3.7
       dist: xenial
+    - env: TOX_SUFFIX="lint"
+      python: 3.8-dev
+      dist: xenial
```
Comment (hugovk) [inline .travis.yml]: A small cleanup suggestion: put `dist: xenial` on line 1 and delete all these.

Comment (hugovk) [reply .travis.yml]: And see https://github.com/neozenith/vcrpy/pull/5 for some more suggestions.

Comment (neozenith) [reply .travis.yml]: Awesome! Thanks for the review, the PR and the help with that. 

I'm kind of new to travis and tox and how they calculate the testing matrix but also how to specify one cell in that matrix too.

Any advice on where to read up about that or just comes with experience and tinkering?

Comment (hugovk) [reply .travis.yml]: You're welcome!

It's generally experience and tinkering, you can have a look through these docs:
https://docs.travis-ci.com/user/build-matrix/
https://docs.travis-ci.com/user/customizing-the-build/#build-matrix

The 3.7/Xenial trick I knew from these alternatives https://github.com/travis-ci/travis-ci/issues/9815.

**Comment by neozenith** [comment]
Merging this, as it achieves the goal of adding the tests for 3.8. Going to address the Cython failures in a different PR.

**Pull request merged**